locks/dynamodb/dynamo_lock_item.go

```diff
@@ -108,15 +109,17 @@ func createItemAttributes(itemId string, client *dynamodb.DynamoDB) (map[string]
 	}, nil
 }
 
-// Return the UserID
+// Return the UserName
 func getCallerIdentity(client *dynamodb.DynamoDB) (string, error) {
 	stsconn := sts.New(session.New(), &client.Config)
 	output, err := stsconn.GetCallerIdentity(&sts.GetCallerIdentityInput{})
 	if err != nil {
 		return "", errors.WithStackTrace(err)
 	}
 
-	return *output.UserId, nil
+	userArn := *output.Arn
+	userName := strings.Split(userArn, "/")[1]
```

Comment:
This risks a panic if the ARN doesn't have any slashes in it, which is possible for some type of ARNs: http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#genref-arns

Perhaps instead of trying to extract the user name, why not store the entire ARN?